src/modules-load/modules-load.c

```diff
@@ -1,28 +1,77 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
 #include <getopt.h>
-#include <sys/stat.h>
+#include <limits.h>
+#include <pthread.h>
+#include <signal.h>
+#include <sys/socket.h>
+#include <unistd.h>
 
 #include "alloc-util.h"
+#include "assert.h"
 #include "build.h"
 #include "conf-files.h"
 #include "constants.h"
 #include "errno-util.h"
 #include "fd-util.h"
 #include "fileio.h"
 #include "log.h"
+#include "macro.h"
 #include "main-func.h"
 #include "module-util.h"
+#include "ordered-set.h"
+#include "parse-util.h"
 #include "pretty-print.h"
 #include "proc-cmdline.h"
 #include "string-util.h"
 #include "strv.h"
 
+#define MODULE_NAME_MAX_LEN (4096UL)
+
 static char **arg_proc_cmdline_modules = NULL;
 static const char conf_file_dirs[] = CONF_PATHS_NULSTR("modules-load.d");
 
 STATIC_DESTRUCTOR_REGISTER(arg_proc_cmdline_modules, strv_freep);
 
+static int modules_list_append_suffix(OrderedSet **module_set, char *modp) {
+        _cleanup_free_ char *mod = modp;
+        int r;
+
+        assert(module_set);
+        assert(mod);
+
+        if (strlen(mod) > MODULE_NAME_MAX_LEN)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Module name max length exceeded (%lu): %s",
+                                       MODULE_NAME_MAX_LEN, mod);
```

Comment:
this function now logs on some errors (this one here) but not on others (i.e. enomem below). we generally want that functions either log about all errors or about none (in which case the caller can do that). [debug level logging is always ok, and not considered logging in this sense].

typically more "library-like" functions should be of the non-logging kind, and "main-function-like" functions should be of the logging kind. the line between the two is usually not clear, so it's important you make a choice and stick to it.